dissect/target/plugins/os/windows/dpapi/dpapi.py

```diff
@@ -54,127 +47,150 @@ def syskey(self) -> bytes:
-    def _get_lsa_key(self) -> bytes:
-        if self._newer_than_vista:
+    @cached_property
+    def lsakey(self) -> bytes:
+        if self._vista_or_newer:
             policy_key = "PolEKList"
         else:
             policy_key = "PolSecretEncryptionKey"
 
-        encrypted_key = (
-            self.target.registry.key(self.SECURITY_KEY.format(KEY=policy_key)).value(self.DEFAULT_REG_VALUE).value
-        )
+        encrypted_key = self.target.registry.key(self.SECURITY_POLICY_KEY).subkey(policy_key).value("(Default)").value
 
-        if self._newer_than_vista:
+        if self._vista_or_newer:
             lsa_key = _decrypt_aes(encrypted_key, self.syskey)
             lsa_key = lsa_key[68:100]
         else:
-            md5 = hashlib.md5()
-            md5.update(self.syskey)
+            ctx = hashlib.md5()
+            ctx.update(self.syskey)
+
+            tmp = encrypted_key[60:76]
             for _ in range(1000):
-                md5.update(encrypted_key[60:76])
-            rc4key = md5.digest()
+                ctx.update(tmp)
 
-            rc4 = ARC4.new(rc4key)
-            lsa_key = rc4.decrypt(encrypted_key[12:60])
+            cipher = ARC4.new(ctx.digest())
+            lsa_key = cipher.decrypt(encrypted_key[12:60])
             lsa_key = lsa_key[16:32]
+
         return lsa_key
 
     @cached_property
     def secrets(self) -> dict[str, bytes]:
-        res = {}
-        reg_secrets = self.target.registry.key(self.SECURITY_KEY.format(KEY="Secrets"))
+        result = {}
+
+        reg_secrets = self.target.registry.key(self.SECURITY_POLICY_KEY).subkey("Secrets")
         for subkey in reg_secrets.subkeys():
-            enc_data = subkey.subkey("CurrVal").value(self.DEFAULT_REG_VALUE).value
-            if self._newer_than_vista:
-                secret = _decrypt_aes(enc_data, self._get_lsa_key())
+            enc_data = subkey.subkey("CurrVal").value("(Default)").value
+            if self._vista_or_newer:
+                secret = _decrypt_aes(enc_data, self.lsakey)
             else:
-                secret = __decrypt_des(enc_data[0xC:], self.syskey)
-            res[subkey.name] = secret
-        return res
+                secret = _decrypt_des(enc_data[12:], self.syskey)
+            result[subkey.name] = secret
+
+        return result
 
     @cached_property
     def master_keys(self) -> dict[str, dict[str, MasterKeyFile]]:
         # This assumes that there is no user named System.
         # As far as I can tell, the name "System" is saved for the actual System user
         # Therefore the user can't actually exist in `all_with_home`
-        user_masterkeys = {self.SYSTEM_USERNAME: {}}
-        for dir in [self.SYSTEM_MASTER_KEY_PATH, self.SYSTEM_USER_MASTER_KEY_PATH]:
-            user_mks = self.load_masterkeys_from_path(
-                self.SYSTEM_USERNAME,
-                dir,
-            )
-            user_masterkeys[self.SYSTEM_USERNAME].update(user_mks)
+        result = {"System": {}}
```

Comment:
You left `SYSTEM_USERNAME` but did not use it here, is that on purpose?

Reply:
Nope, accidental.